src/components/Buttons/ClipboardButton.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState, ReactNode } from 'react'
+import Popper from 'popper.js'
+import { Button, Popover, PopoverHeader, PopoverBody } from 'reactstrap'
+import { copyToClipboard } from '../../helpers/copyToClipboard'
+
+interface ClipboardButtonProps {
+  disabled: boolean
+  children: ReactNode
+  textToCopy?: string
+  popoverBody: ReactNode
+  popoverHeader?: string
+  popoverPlacement?: Popper.Placement
+  popoverTarget: string | HTMLElement | React.RefObject<HTMLElement>
+}
+
+const checkPopoverTarget = (popoverTarget: string | HTMLElement | React.RefObject<HTMLElement>) => {
+  if (typeof popoverTarget === 'string') {
+    let element = document.getElementById(popoverTarget)
+
+    if (!element) {
+      element = document.createElement('div')
+      element.setAttribute('id', popoverTarget)
+      document.body.append(element)
+    }
+  }
+}
+
+export default function ClipboardButton({
+  disabled,
+  children,
+  textToCopy,
+  popoverBody = '',
+  popoverHeader = '',
+  popoverPlacement = 'bottom',
+  popoverTarget,
+}: ClipboardButtonProps) {
+  const [popoverOpen, setPopoverOpen] = useState(false)
+  const closePopover = () => setPopoverOpen(false)
+  const onClick = () => {
+    if (textToCopy) {
+      copyToClipboard(String(textToCopy))
+      setPopoverOpen(true)
+    }
+  }
+
+  useEffect(() => {
+    if (popoverTarget) {
+      checkPopoverTarget(String(popoverTarget))
+    }
+  }, [popoverTarget])
+
+  return (
+    <>
+      <Button disabled={disabled} onClick={onClick} color="primary" size="sm">
+        {children}
+      </Button>
+      <Popover placement={popoverPlacement} open={popoverOpen} target={popoverTarget} ontoggle={closePopover}>
+        <PopoverHeader>{popoverHeader}</PopoverHeader>
+        <PopoverBody>{popoverBody}</PopoverBody>
+      </Popover>
```

Comment:
Doesn't show up (but doesn't break the app either 😄 ).

Anyone knows why?

Reply:
Because we are already in a popover? I think this particular implementation is modal, that means there could be only one.

Maybe we could just change the button's text and/or add a green checkmark somewhere?